### DISCUSSION TITLE
Adding voice notes

### DISCUSSION THREAD

**Discussion by nicoKoehler** [usememos/memos#2976] [Ideas]
Dear team & community

First off, thank you that this app exists in the first place. It is the very reason I got started with self hosting, as my prior favorite diary app _Diarium_ does not support linux and does not have a web app. 

Anyways, so I use _Memos_ as a diary app. Regardless of that, it would be incredibly helpful if one could add voice notes to notes, along the same lines as pictures. I am not sure if recording VMs is feasible, but at least being able to attach them to a voicenote, and optionally even play them natively, that would be a huge help.

What do you think? Is this feasible? Desired? Or pointless? 

(my current work around is to record VMs on my phone, sync them to my file server, and then add the file link to my notes in _Memos_. It's a bit of a journey...)

**Comment by lincolnthalles** [comment]
You already can upload audio files and play them, as long the container and codecs are [supported by browsers](https://www.chromium.org/audio-video/).
![image](https://github.com/usememos/memos/assets/7476810/d17df054-05aa-49ca-82b1-a8de7872fbf0)

Direct audio recording would be an awesome feature. The [MediaRecorder API](https://developer.mozilla.org/en-US/docs/Web/API/MediaRecorder) can do that. It would be a matter of creating the interface and attaching the recorded file as a resource.

I suggest you create an enhancement request in [Memos's issues](https://github.com/usememos/memos/issues).

**Reply by nicoKoehler** [reply]
@lincolnthalles  how did you just upload that voicenote? I now that browsers generally can play audio content, but I have not figured out how to add it to memos notes

**Reply by lincolnthalles** [reply]
Just click on the image icon. You can upload whatever type of file you want. If it's video or audio, the browser will display a basic player.

![image](https://github.com/usememos/memos/assets/7476810/88fe8f9e-22de-44bb-b508-d2e1e1311bb2)

**Reply by nicoKoehler** [reply]
holy smokes... never thought of trying this. THis is absolutely brilliant, thank you so much!
